syntax = "proto3";
package rpc;

message SimpleRequest {
  oneof payload {
    int32 int_value = 1;
    string str_value = 2;
  }
}

message SimpleResponse {
  oneof payload {
    int32 int_value = 1;
    string str_value = 2;
  }
}

message StreamRequest {
  int32 count = 1;
}

message StreamResponse {
  int32 value = 1;
}

service RPCService {
  rpc SimpleCall(SimpleRequest) returns (SimpleResponse);
  rpc StreamValues(StreamRequest) returns (stream StreamResponse);
}
